### PR TITLE
Update CLI configuration handling

### PR DESCRIPTION
**Pull request by iramiller**
## Description

Wires up the new client config with config file into cli context.

closes: #394

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: cmd/provenanced/cmd/clientconfig.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		case flags.FlagNode:
+			conf.SetNode(value)
+		case flags.FlagBroadcastMode:
+			conf.SetBroadcastMode(value)
+		default:
+			return errUnknownConfigKey(key)
+		}
+
+		confFile := filepath.Join(configPath, "client.toml")
+		if err := config.WriteConfigToFile(confFile, conf); err != nil {
```
Comment (dwedul-figure) [inline cmd/provenanced/cmd/clientconfig.go:87]: Nit: It'd be nice if `"client.toml"` were only defined in one place. It's also defined in `config/client.go#ReadFromClientConfig`. I'm not really sure where best to set it, though, so just kind of talking I guess.

File: cmd/provenanced/config/toml.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+	"github.com/spf13/viper"
+)
+
+const defaultConfigTemplate = `# This is a TOML config file.
+# For more information, see https://github.com/toml-lang/toml
+
+###############################################################################
+###                           Client Configuration                            ###
+###############################################################################
```
Comment (dwedul-figure) [inline cmd/provenanced/config/toml.go:17]: Super nit: right alignment. :eye-twitch:

Comment (iramiller) [reply cmd/provenanced/config/toml.go:17]: most of the stuff in the config folder was directly from the SDK upstream... could use some polish

**Review by dwedul-figure — APPROVED** [review]
LGTM

File: cmd/provenanced/cmd/clientconfig.go (hunk 1 above)
Comment (iramiller) [reply cmd/provenanced/cmd/clientconfig.go:87]: The config/client.go is copied directly from the SDK and not modified other than the keyring backend default that had to be changed to allow our chain development networks to function.  I would rather not refactor it too much.  We could restructure it more in #403 if needed though.

**Comment by iramiller** [comment]
We will do some additional refactor work on the cmd structure to streamline the config as part of #403.

**Pull request merged**